### PR TITLE
Unify the config entry to avoid multiple load config

### PR DESCRIPTION
**Pull request by luxiaba**

### CODE REVIEW

File: tests/test_config.py [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
-    load_config(CONFIG_FILE)
+    LBConfig.load(CONFIG_FILE)
 
 
 def test_load_providers_from_config():
-    config = load_config(CONFIG_FILE)
-    providers = load_providers(config.provider, PARENT_DIR.parent / "dev_datadir")
-    assert len(providers) > 0
+    config = LBConfig.load(CONFIG_FILE)
+    assert len(config.provider_cls) > 0
```
Comment (messense) [inline tests/test_config.py:15]: This test slightly changed behavior, previous it will actually initialize the provider class in `load_providers`.

Comment (luxiaba) [reply tests/test_config.py:15]: Yeah, current test state is not perfect, and this PR seems to reduce the test cov. I'll improve the test next time.

File: lobbyboy/utils.py [hunk 2]
```diff
@@ -273,3 +274,12 @@ def try_load_key_from_file(
     if not (pri_key and pub_key) and raise_error:
         raise FileNotFoundError(f"ssh key pair not found: {private_key_file}/{public_key_file}")
     return pri_key, pub_key
+
+
+def get_cls(cls_path: str = ""):
```
Comment (messense) [inline lobbyboy/utils.py:279]:
```suggestion
def import_class(cls_path: str = ""):
```

File: lobbyboy/config.py [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
 from json import JSONDecodeError
 from pathlib import Path
-from typing import Dict, List, Tuple, OrderedDict as typeOrderedDict, Optional
+from typing import Dict, List, Tuple, OrderedDict as typeOrderedDict, Optional, Type, Any
 
 import toml
 
 from lobbyboy.exceptions import InvalidConfigException
-from lobbyboy.utils import lb_dict_factory, confirm_dc_type
+from lobbyboy.utils import lb_dict_factory, confirm_dc_type, get_cls
```
Comment (messense) [inline lobbyboy/config.py:13]:
```suggestion
from lobbyboy.utils import lb_dict_factory, confirm_dc_type, import_class
```

File: lobbyboy/config.py [hunk 4]
```diff
@@ -99,84 +127,73 @@ def validate(self) -> Tuple[bool, Optional[str]]:
...
-    def reload(self) -> "LBConfig":
-        return load_config(self._file)
+        # Initialize the configuration with each provider's own config class.
+        config: Dict
+        for name, config in self.provider.items():
+            if not config.get("enable", True):
+                continue
+
+            load_module = config.get("load_module", "")
+            provider_cls = get_cls(load_module)
```
Comment (messense) [inline lobbyboy/config.py:143]:
```suggestion
            provider_cls = import_class(load_module)
```

File: lobbyboy/utils.py (hunk 2 above)
Comment (messense) [reply lobbyboy/utils.py:279]: Looks like this is reverted?

Comment (luxiaba) [reply lobbyboy/utils.py:279]: seems so, let it be, i'll commit it later.

**Review by messense — APPROVED** [review]

**Pull request merged**